### ISSUE TITLE
RichText Hyperlink segments do not wrap

### ISSUE THREAD

**Issue by Arsen6331** [fyne-io/fyne#3393]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When using a hyperlink segment in RichText, it does not wrap the text, despite `TextWrapWord` being set. Normal text does wrap.

### How to reproduce

1. Create a RichText widget with a hyperlink long enough that it should get wrapped.
2. The hyperlink does not get wrapped.

### Screenshots

With hyperlink:

![image](https://user-images.githubusercontent.com/41810442/201426362-1223e8cb-d090-4298-b0c0-1a80e5d1fa60.png)

Without hyperlink:

![image](https://user-images.githubusercontent.com/41810442/201426445-f20cba81-0851-43f3-8756-f67dd2df1c97.png)

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("test")

	rt := widget.NewRichTextFromMarkdown(`Lorem [ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore](https://example.com) et dolore magna aliqua. Porttitor leo a diam sollicitudin tempor. Nunc faucibus a pellentesque sit amet porttitor eget.`)
	rt.Wrapping = fyne.TextWrapWord

	w.SetContent(rt)
	w.ShowAndRun()
}
```

### Fyne version

2.2.4

### Go compiler version

1.19.3

### Operating system

Linux

### Operating system version

Arch Linux

**Comment by d4x1** [comment]
I run this code and it can be reproduced. I think it may be caused by [lineBounds](https://github.com/fyne-io/fyne/blob/develop/widget/richtext.go#L767) in richtext.

@andydotxyz Do you have any design doc for richtext?  It will be easier to fix it if there are more details.

**Comment by andydotxyz** [comment]
> @andydotxyz Do you have any design doc for richtext? It will be easier to fix it if there are more details.

The high level proposal is at https://github.com/fyne-io/proposals/blob/main/richtext.md.

The problem is fairly complex, because we're not just adding wrap to hyperlink, but asking it to wrap with an initial offset as well (when the hyperlink wraps on a line it was not the first item of).
There are some `if _, ok; obj.(*canvas.Text); ok` in the rich text code as we assumed only text would wrap.
Potentially we need to split hyperlink into two different objects when rendering to simulate the wrap?

**Comment by d4x1** [comment]
> > @andydotxyz Do you have any design doc for richtext? It will be easier to fix it if there are more details.
> 
> The high level proposal is at https://github.com/fyne-io/proposals/blob/main/richtext.md.
> 
> The problem is fairly complex, because we're not just adding wrap to hyperlink, but asking it to wrap with an initial offset as well (when the hyperlink wraps on a line it was not the first item of). There are some `if _, ok; obj.(*canvas.Text); ok` in the rich text code as we assumed only text would wrap. Potentially we need to split hyperlink into two different objects when rendering to simulate the wrap?


Agreed. Splitting hyperlink into some objects is a practical way,  another problem is when the mouse cursor is hovering it,  these split objects should share the same an underline.  
In short, implementing a rich text it is really very trivial.

**Comment by ablegao** [comment]
Is there a good solution available at the moment?

**Comment by andydotxyz** [comment]
This is an open bug, until it is fixed there isn't a fix.
You could use shorter text in your hyperlinks I guess?

**Comment by ablegao** [comment]
I'm developing an epub reader and some of the books have a problem with excessively long links.

**Comment by andydotxyz** [comment]
Thanks for the fix @MaxGyver83, this is now on develop and will be in the next release

**Comment by dweymouth** [comment]
There is still an issue that can be treated as a separate bug. Hyperlinks wrap now but when wrapping across N lines it looks and behaves as N separate hyperlinks. We need a way to connect the hover of any of the wrapped segments so that they all underline in unison.
